src/ArrayInterface.jl

```diff
@@ -1,735 +1,18 @@
-
-function Base.strides(A::AbstractArray2)
-    defines_strides(A) && return map(Int, ArrayInterface.strides(A))
-    throw(MethodError(Base.strides, (A,)))
-end
-Base.strides(A::AbstractArray2, dim) = Int(ArrayInterface.strides(A, dim))
-
-function Base.IndexStyle(::Type{T}) where {T<:AbstractArray2}
-    if parent_type(T) <: T
-        return IndexCartesian()
-    else
-        return IndexStyle(parent_type(T))
-    end
-end
-
-function Base.length(A::AbstractArray2)
-    len = known_length(A)
-    if len === nothing
-        return Int(prod(size(A)))
-    else
-        return Int(len)
-    end
-end
-
-@propagate_inbounds Base.getindex(A::AbstractArray2, args...) = getindex(A, args...)
-@propagate_inbounds Base.getindex(A::AbstractArray2; kwargs...) = getindex(A; kwargs...)
-
-@propagate_inbounds function Base.setindex!(A::AbstractArray2, val, args...)
-    return setindex!(A, val, args...)
-end
-@propagate_inbounds function Base.setindex!(A::AbstractArray2, val; kwargs...)
-    return setindex!(A, val; kwargs...)
-end
-
-
-"""
-    is_lazy_conjugate(::AbstractArray) -> Bool
-
-Determine if a given array will lazyily take complex conjugates, such as with `Adjoint`. This will work with
-nested wrappers, so long as there is no type in the chain of wrappers such that `parent_type(T) == T`
-
-Examples
-
-    julia> a = transpose([1 + im, 1-im]')
-    2×1 transpose(adjoint(::Vector{Complex{Int64}})) with eltype Complex{Int64}:
-     1 - 1im
-     1 + 1im
-
-    julia> ArrayInterface.is_lazy_conjugate(a)
-    True()
-
-    julia> b = a'
-    1×2 adjoint(transpose(adjoint(::Vector{Complex{Int64}}))) with eltype Complex{Int64}:
-     1+1im  1-1im
-
-    julia> ArrayInterface.is_lazy_conjugate(b)
-    False()
-
-"""
-is_lazy_conjugate(::T) where {T <: AbstractArray} = _is_lazy_conjugate(T, False())
-is_lazy_conjugate(::AbstractArray{<:Real})  = False()
-
-function _is_lazy_conjugate(::Type{T}, isconj) where {T <: AbstractArray}
-    Tp = parent_type(T)
-    if T !== Tp
-        _is_lazy_conjugate(Tp, isconj)
-    else
-        isconj
-    end
-end
-
-function _is_lazy_conjugate(::Type{T}, isconj) where {T <: Adjoint}
-    Tp = parent_type(T)
-    if T !== Tp
-        _is_lazy_conjugate(Tp, !isconj)
-    else
-        !isconj
-    end
-end
-
-include("ranges.jl")
-include("axes.jl")
-include("size.jl")
-include("dimensions.jl")
-include("indexing.jl")
-include("stridelayout.jl")
-include("broadcast.jl")
+    permute, invariant_permutation, field_type, reduce_tup
 
 function __init__()
-
-    @require SuiteSparse = "4607b0f0-06f3-5cda-b6b1-a6196a1729e9" begin
-        function lu_instance(jac_prototype::SparseMatrixCSC)
-            return SuiteSparse.UMFPACK.UmfpackLU(
-                Ptr{Cvoid}(),
-                Ptr{Cvoid}(),
-                1,
-                1,
-                jac_prototype.colptr[1:1],
-                jac_prototype.rowval[1:1],
-                jac_prototype.nzval[1:1],
-                0,
-            )
-        end
```

Comment:
Where did this one go?

Reply:
It's part of the standard library so I put it in ArrayInterfaceCore

Reply:
Probably better to just depend on SuiteSparse than depend on Requires.

Reply:
A quick look at the code/Project.toml shows you're depending on SparseArrays but not Requires.